### PR TITLE
feat(iot-sitewise): add bulk import, buffered ingestion tools, and prompts to iot-sitewise

### PR DESCRIPTION
**Pull request by ashuanand1226**
# Add Bulk Import and Buffered Ingestion tools and prompts to AWS IoT SiteWise MCP Server

### Changes

This PR adds comprehensive bulk data import capabilities to the AWS IoT SiteWise MCP Server, including:

- **5 new bulk import tools**: `create_bulk_import_job`, `create_buffered_ingestion_job`, `create_bulk_import_iam_role`, `list_bulk_import_jobs`, and `describe_bulk_import_job`
- **New bulk import workflow prompt**: Step-by-step guidance for setting up bulk data ingestion with CSV validation, IAM role creation, and job monitoring
- **Enhanced validation**: CSV file format validation, job ID format checking, and bulk import parameter validation
- **Comprehensive test coverage**: New test cases covering all bulk import scenarios and edge cases
- **Updated documentation**: README updated with new tools and usage examples

### User experience

**Before**: Users could only ingest data one batch at a time using `batch_put_asset_property_value`, limiting them to small-scale data ingestion scenarios.

**After**: Users can now:
- Ingest historical or buffered data use SiteWise Bulk Import capabilities
- Get step-by-step guidance through the bulk import workflow prompt
- Automatically create required IAM roles with proper S3 permissions for bulk import and buffered ingestion jobs
- Monitor job progress and list all bulk import jobs for the user.

## Checklist

* [x] I have reviewed the [contributing guidelines](https://github.com/awslabs/mcp/blob/main/CONTRIBUTING.md)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented

Is this a breaking change? (Y/N): **N**

**RFC issue number**: N/A

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the [project license](https://github.com/awslabs/mcp/blob/main/LICENSE).

### CODE REVIEW

**Review by ychamare — DISMISSED** [review]
approved, code has bee also reviewed internally before PR

**Review by ychamare — APPROVED** [review]

File: src/aws-iot-sitewise-mcp-server/awslabs/aws_iot_sitewise_mcp_server/tools/sitewise_data.py [hunk 1]
```diff
@@ -31,6 +33,310 @@
...
+        description='List of files in Amazon S3 that contain your data. Each file should have "bucket", "key", and optionally "versionId" fields',
+    ),
+    error_report_location: Dict[str, str] = Field(
+        ..., description='Amazon S3 destination for errors. Must have "bucket" and "prefix" fields'
+    ),
+    job_configuration: Dict[str, Any] = Field(
+        ...,
+        description='Job configuration including file format. For CSV: {"fileFormat": {"csv": {"columnNames": ["ALIAS", "ASSET_ID", ...]}}}',
+    ),
+    adaptive_ingestion: bool = Field(
```
Comment (charlie-7) [inline src/aws-iot-sitewise-mcp-server/awslabs/aws_iot_sitewise_mcp_server/tools/sitewise_data.py:56]: Considering `adaptiveIngestion` is an optional field, should we explicitly set its default value to `false` when it's not provided? This would help prevent the agent from enabling it (`true`) unintentionally.

File: src/aws-iot-sitewise-mcp-server/awslabs/aws_iot_sitewise_mcp_server/tools/sitewise_data.py [hunk 2]
```diff
@@ -856,13 +1357,108 @@ def execute_query(
...
+
+        # Create policy for S3 permissions
+        data_bucket_resources = []
+        for bucket in data_bucket_names:
+            data_bucket_resources.extend([f'arn:aws:s3:::{bucket}', f'arn:aws:s3:::{bucket}/*'])
+
+        policy_document = {
+            'Version': '2012-10-17',
+            'Statement': [
+                {'Effect': 'Allow', 'Action': ['s3:*'], 'Resource': data_bucket_resources},
```
Comment (charlie-7) [inline src/aws-iot-sitewise-mcp-server/awslabs/aws_iot_sitewise_mcp_server/tools/sitewise_data.py]: We should scope down the policies to align with those defined in our public documentation: https://docs.aws.amazon.com/iot-sitewise/latest/userguide/CreateBulkImportJob.html

Similar for error bucket actions.

File: src/aws-iot-sitewise-mcp-server/awslabs/aws_iot_sitewise_mcp_server/tools/sitewise_data.py [hunk 3]
```diff
@@ -31,6 +33,310 @@
...
+    except ClientError as e:
+        return {
+            'success': False,
+            'error': str(e),
+            'error_code': e.response['Error']['Code'],
+        }
+
+
+@tool_metadata(readonly=False)
+def create_buffered_ingestion_job(
```
Comment (charlie-7) [inline src/aws-iot-sitewise-mcp-server/awslabs/aws_iot_sitewise_mcp_server/tools/sitewise_data.py:268]: Not sure if this tool is needed — it only sets adaptive_ingestion to true. Since we’ve already defined when to use this in the description of the create_bulk_import_job tool, I think this may not be necessary.

File: src/aws-iot-sitewise-mcp-server/awslabs/aws_iot_sitewise_mcp_server/tools/sitewise_data.py (hunk 1 above)
Comment (ashuanand1226) [reply src/aws-iot-sitewise-mcp-server/awslabs/aws_iot_sitewise_mcp_server/tools/sitewise_data.py:56]: For the BI tool, I wanted the customer to explicitly let us know if adaptive ingestion should be enabled or not. That is why I haven't specified a default value. If we want it to adhere to our API we can set this to default to false though.

File: src/aws-iot-sitewise-mcp-server/awslabs/aws_iot_sitewise_mcp_server/tools/sitewise_data.py (hunk 2 above)
Comment (ashuanand1226) [reply src/aws-iot-sitewise-mcp-server/awslabs/aws_iot_sitewise_mcp_server/tools/sitewise_data.py]: Yea, that's a good point. I had set this to the same as our public documentation but seems during the iterative process of testing and development, Q decided to expand the actions allowed.

File: src/aws-iot-sitewise-mcp-server/awslabs/aws_iot_sitewise_mcp_server/tools/sitewise_data.py (hunk 3 above)
Comment (ashuanand1226) [reply src/aws-iot-sitewise-mcp-server/awslabs/aws_iot_sitewise_mcp_server/tools/sitewise_data.py:268]: We explicitly wanted to create two different tools for BI and Buffered ingestion as they are advertised as two different features to the customer

**Review by ychamare — APPROVED** [review]

**Review by ychamare — APPROVED** [review]
